src/openssl.c

```diff
@@ -11975,6 +11975,47 @@ static int cipher_final(lua_State *L) {
 } /* cipher_final() */
 
 
+static int cipher_get_tag(lua_State *L) {
```

Comment:
Should this be `cipher_gcm_get_tag` or similar to follow the naming of the option?

Reply:
Possibly. It's actually not unique to GCM, but to the family of AEAD modes - in OpenSSL this includes CCM. In 1.1.x there is now therefore an alias for `EVP_CTRL_AEAD_GET_TAG` but this constant is not defined in 1.0.x. The current code should work with both GCM and CCM.

If we change the name, we could call it e.g. `:get_aead_tag()`, but I don't know whether this level of disambiguation is necessary.

Reply:
Would it be sensible to use `EVP_CIPHER_type` to get the type of the cipher and then use the relevant constant?

Reply:
Could you add a comment saying as much?

Reply:
Added in https://github.com/wahern/luaossl/pull/201/commits/4a8b770fc5bbb7d89f8d59e9c84e13d8091ee4f8